libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/spm/SwiftPackageExtension.kt

```diff
@@ -0,0 +1,380 @@
+    class SupportedPlatform() {
+        var platforms = mutableListOf<SupportedPlatform>()
+
+        lateinit var type: PlatformType
+        lateinit var version: String
+
+        constructor(type: PlatformType, version: String) : this() {
+            this.type = type
+            this.version = version
+        }
+
+        fun macOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.macOS, version)
+            platforms.add(platform)
+        }
+
+        fun iOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.iOS, version)
+            platforms.add(platform)
+        }
+
+        fun tvOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.tvOS, version)
+            platforms.add(platform)
+        }
+
+        fun watchOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.watchOS, version)
+            platforms.add(platform)
+        }
+
+        enum class PlatformType {
+            macOS, iOS, tvOS, watchOS
+        }
+    }
+
+    class Products {
+        var libraries = mutableListOf<ProductType.Library>()
+        var executables = mutableListOf<ProductType.Executable>()
+
+        fun library(name: String, configure: ProductType.Library.() -> Unit) {
+            val library = ProductType.Library(name).apply(configure)
+            libraries.add(library)
+        }
+
+        fun library(name: String, configure: Closure<*>) = library(name) {
+            ConfigureUtil.configure(configure, this)
+        }
+
+        fun executable(name: String, configure: ProductType.Executable.() -> Unit) {
+            val executable = ProductType.Executable(name).apply(configure)
+            executables.add(executable)
+        }
+
+        fun executable(name: String, configure: Closure<*>) = executable(name) {
+            ConfigureUtil.configure(configure, this)
+        }
+
+        sealed class ProductType {
+            class Executable(@Input val name: String) {
+                @Input
+                val targets = arrayListOf<String>()
+
+                operator fun String.unaryPlus() {
+                    targets.add(this)
+                }
+            }
+
+            class Library(@Input val name: String) {
+                // TODO: доделать
+                @Input
+                lateinit var type: LibraryType
+
+                @Input
+                val targets = arrayListOf<String>()
+
+                operator fun String.unaryPlus() {
+                    targets.add(this)
+                }
+
+                enum class LibraryType {
+                    STATIC,
+                    DYNAMIC
+                }
+            }
+        }
+    }
+
+    // TODO: add fun with ranges
+    // TODO: add Package Dependency Requirement
+    class Dependencies {
+        var dependencies = mutableListOf<Package>()
+
+        fun `package`(url: String, version: String) {
```

Comment:
Manual overloading and \`\` quotes are not required. Use default parameters with `@JvmOverloads` annotation